Added a password hashing algorithm and several implementation methods…

connormcdermid opened the pull request:
Although this would be better served by a new branch for password hashing, this seems fine for now. If you have any issues with this code, the glory of git being that you can roll back to a previous version, just ask and I'll work to fix it.
This pull request works to resolve #2.

connormcdermid committed:
Added a password hashing algorithm and several implementation methods into PlayerConnection, which can be moved to a new object if need be at a later date. All of these methods are private.

Also added a .gitignore to exclude my .idea file.

Signed-off-by: Connor McDermid <cam-o-man@live.com>

CAM-o-man commented:
Zach - a summary of how pull requests work: you can take a look at the code I've submitted here. If you like it, you can merge the pull request. Otherwise, you can discuss it on this form and I can make changes, or you can close it altogether.

connormcdermid committed:
Added several exceptions for use serverside:
    A RuntimeException InvalidPasswordException for if a password does not match
    An Exception InvalidSaveFileException for if a save file is mis-formatted or otherwise invalid.

Signed-off-by: Connor McDermid <cam-o-man@live.com>

CAM-o-man commented:
Added several serverside exceptions for use in creating proper failstates, see #3

connormcdermid committed:
Create LICENSE

connormcdermid committed:
Merge pull request #1 from CAM-o-man/docs

Create LICENSE
